nbcelltests/tests_vendored.py

```diff
@@ -54,8 +54,116 @@
 
 import unittest
 
+import nbformat
 from nbval.kernel import RunningKernel
 
+from nbcelltests.shared import is_empty, cell_injected_into_test, source2lines, lines2source, cell_inj_span, get_test
+
+
+def get_kernel(path_to_notebook):
+    notebook = nbformat.read(path_to_notebook, 4)
+    # TODO don't default kernel like this here (address as part of https://github.com/jpmorganchase/nbcelltests/issues/164)
+    return notebook['metadata'].get('kernelspec', {}).get('name', 'python')
+
+
+def _inject_cell_into_test(cell_source, test_source):
+    """Inserts cell_source into test.
+
+    The cell_source is inserted wherever %cell appears at the stripped
+    start of a line.
+
+    All lines of cell_source are prefixed with the same leading
+    whitespace as %cell.
+
+    Example:
+
+      test:
+        if x > 0:
+            %cell # end of %cell
+        assert x == 10
+        assert z == 15
+
+      cell:
+        x*=5
+        z = x + \
+            5
+        print(x)
+
+      celltest:
+        if x > 0:
+            x*=5
+            z = x + \
+                5
+            print(x) # end of %cell
+        assert x == 10
+        assert z == 15
+    """
+    celltest_lines = []
+    for test_line in source2lines(test_source):
+        cell_span = cell_inj_span(test_line)
+        if cell_span is not None:
+            prefix = test_line[0:cell_span[0]]
+            for cell_line in source2lines(cell_source):
+                celltest_lines.append(prefix + cell_line)
+
+            suffix = test_line[cell_span[1]::]
+            if len(suffix) > 0:
+                if len(celltest_lines) == 0:
+                    celltest_lines.append('')
+                celltest_lines[-1] += suffix
+        else:
+            celltest_lines.append(test_line)
+    return lines2source(celltest_lines)
+
+
+def get_celltests(path_to_notebook):
+    """
+    Return a dictionary of {code cell number: celltest} for all the code
+    cells in the given notebook.
+
+    A celltest is a source code string of the test supplied for a
+    cell, plus the cell itself wherever (if) injected into the test
+    using %cell.
+
+    (celltest is actually currently a dictionary of skip_reason (if any)
+    and the source code, but we're unsure if we want to keep the skipping
+    part. If keeping, we'll need to clean up and doc.)
+    """
```

Comment:
```suggestion
    """
    # TODO: Make function and docstring match.
```

Reply:
Actually this TODO comment already appears a little further down.